### PR TITLE
refactor: JDTTreeBuilder

### PR DESCRIPTION
**Pull request by danglotb**
Split into multiple classes, remove unused method and fields for the sake of readability.

### CODE REVIEW

File: src/main/java/spoon/support/compiler/jdt/ParentExiter.java [hunk 1]
```diff
@@ -462,7 +462,6 @@ public void visitCtPackage(CtPackage ctPackage) {
 				ctPackage.getTypes().remove(child);
 			}
 			ctPackage.getTypes().add((CtType<?>) child);
-			this.jdtTreeBuilder.context.addCreatedType((CtType<?>) child);
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/compiler/jdt/ParentExiter.java]: Never used?

Comment (danglotb) [inline src/main/java/spoon/support/compiler/jdt/ParentExiter.java]: This method appends the given CtType to a list in BuilderContext and this list is never used.

Comment (GerardPaligot) [inline src/main/java/spoon/support/compiler/jdt/ParentExiter.java]: Ok :)

File: src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java [hunk 2]
```diff
@@ -911,6 +828,8 @@ public void endVisit(WhileStatement whileStatement, BlockScope scope) {
 		context.exit(whileStatement);
 	}
 
+
+	//TODO
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java]: ?

File: src/main/java/spoon/support/compiler/jdt/HelperJDTTreeBuilder.java [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+		}
+		return null;
+	}
+
+	/**
+	 * Check if the annotation is declared with the given element type.
+	 *
+	 * @param a           An annotation.
+	 * @param elementType Type of the annotation.
+	 * @return true if the annotation is compitble with the given element type.
```
Comment (seintur) [inline src/main/java/spoon/support/compiler/jdt/HelperJDTTreeBuilder.java]: compatible (not compitble)

File: src/main/java/spoon/support/compiler/jdt/HelperJDTTreeBuilder.java [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+import spoon.reflect.declaration.CtAnnotatedElementType;
+import spoon.reflect.declaration.CtAnnotation;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtTypedElement;
+import spoon.reflect.declaration.ModifierKind;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+class HelperJDTTreeBuilder {
```
Comment (monperrus) [inline src/main/java/spoon/support/compiler/jdt/HelperJDTTreeBuilder.java]: JDTTreeBuilderHelper

**Pull request merged**